staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1030,13 +1030,14 @@ func (e *Store) DeleteCollection(ctx context.Context, deleteValidation rest.Vali
 					errs <- err
 					return
 				}
-				if _, _, err := e.Delete(ctx, accessor.GetName(), deleteValidation, options); err != nil && !kubeerr.IsNotFound(err) {
+				delOptions := options.DeepCopy()
+				if _, _, err := e.Delete(ctx, accessor.GetName(), deleteValidation, delOptions); err != nil && !kubeerr.IsNotFound(err) {
 					klog.V(4).Infof("Delete %s in DeleteCollection failed: %v", accessor.GetName(), err)
 					errs <- err
 					return
 				}
 			}
-		}()
+		}(options)
```

Comment:
i've just realized that we should `DeepCopy()` here, not inside `go routines`, since we don't know which one is going to be scheduled first. WDYT ?

Reply:
DeepCopy doesn't modify options, so it should be fine to call concurrently inside the goroutine

Reply:
but `Delete` method could, thus each `go routine` should get its own copy of `options`

Reply:
correct, so passing options.DeepCopy() to Delete() makes sense (xref https://github.com/kubernetes/kubernetes/pull/81288#discussion_r313369062)

Reply:
ah, now I see it, `options` can be modified since each `Delete` function gets a copy of it anyway :)